action.yaml

```diff
@@ -282,6 +318,89 @@ runs:
           esac
         }
 
+        # Secure credential handling functions
+        # These functions use process substitution to keep credentials
+        # out of logs, process lists, and debug output
+
+        # Function to securely upload a file using curl --config
+        secure_upload_file() {
+          local file="$1"
+          local upload_url="$2"
+          local content_type="${3:-application/octet-stream}"
+
+          # Use process substitution to keep credentials secure
+          curl -s -f --max-redirs 0 -w '%{http_code}\n' \
+            --config <(
+              printf 'user = "%s:%s"\n' "$nexus_user" "$nexus_pass"
```

Comment:
Are you not able to use a `~/.netrc` file with the following information in in?

```
machine $nexus_base login $nexus_user password $nexus_pass
```

Writing the file out once and telling curl to use the netrc file (the flag is `-n`) would be safer than having this code repeated multiple times as now if there is a security issue with how it's being written out it has to be validated multiple times across the script instead of once.